src/ray/raylet/task_util.h

```diff
@@ -0,0 +1,77 @@
+#ifndef RAY_RAYLET_TASK_UTIL_H
+#define RAY_RAYLET_TASK_UTIL_H
+
+#include "ray/protobuf/common.pb.h"
+#include "ray/raylet/task_spec.h"
+
+namespace ray {
+
+namespace raylet {
+
+/// Helper function for building the common fields of a `TaskSpec` message.
+/// See `common.proto` for the documentation of the arguments.
+///
+/// \param[out] message The `TaskSpec` message to build.
+void BuildCommonTaskSpec(
+    rpc::TaskSpec *message, const Language &language,
```

Comment:
The C++ style guide says

>When ordering function parameters, put all input-only parameters before any output parameters. In particular, do not add new parameters to the end of the function just because they are new; place new input-only parameters before the output parameters.

Same with the other functions here.

https://google.github.io/styleguide/cppguide.html#Output_Parameters

Reply:
I thought about this. But there are also a few benefits of putting `message` as the first parameter. E.g.,
1. It's very clear that the `message` argument is the one that we're building. 
2. It allows making certain input arguments optional with default values. 

Let me investigate if there's a better solution.

Reply:
I thought about this. But there are also a few benefits of putting `message` as the first parameter. E.g.,
1. It's very clear that the `message` argument is the one that we're building. 
2. It allows making certain input arguments optional with default values. 

Let me investigate if there's a better solution.

Reply:
I changed these helper functions to a builder class. The issue doesn't exist now.

Reply:
@raulchen I'm confused, I don't see the builder class, but I still see these functions with the `rpc::TaskSpec *message` argument.

Reply:
@robertnishihara did you refresh the page? the builder class is here https://github.com/ray-project/ray/pull/5121/files#diff-713a8650b1cf28f9e78ca583e1ae35aeR11